### DISCUSSION TITLE
Going idle if no LED device reachable?

### DISCUSSION THREAD

**Discussion by krisrok** [awawa-dev/HyperHDR#709] [Q&A]
Hi, my setup is an ESP8266 running WLED which I can turn on and off via a simple power switch behind my display. 

Maybe I just did not find the right setting yet. But the HyperHDR process on my windows machine produces constant load on GPU and CPU even if there is no active connection to the WLED device (because it's off). 

With other similar projects like Prismatik or Luciferin they go into an idle mode if they don't have an active connection to a device. This is pretty convenient as the service can always be running in the background: It only enables the grabbing/processing if there is someone listening.

**Comment by awawa-dev** [comment]
Hi
In version 20 there is a new option (https://github.com/awawa-dev/HyperHDR/issues/490) that can disable the grabber if no light is active: disabled by users or due to connection drop. But WLED uses UDP which rarely and depending on the operating system throws an exception which the application can detect... sudden powering down the esp8266 is usually not enough, a more serious problem such as an unavailable host network is usually required to force the problem to be signaled.

**Reply by krisrok** [reply]
I just checked and this only applies to USB grabbers but not the software grabber. Could maybe be ported over easily?

Regarding the UDP "connection", it would likely need special handling via other means and depending on the specific protocol, e.g. adding some kind of heartbeat via REST in case of WLED...?

**Reply by awawa-dev** [reply]
> I just checked and this only applies to USB grabbers but not the software grabber. Could maybe be ported over easily?

Yes, it can be easily adopted. On the other hand: WLED on esp8266 equipped with a power switch and all handled by stateless UDP, looks more like a custom solution than a standard one. And for non-standard user setups, we rather have JSOP API (or MQTT). For example: a system script that pings a target and simply executes a GET command locally to enable or disable the system grabber.

**Reply by krisrok** [reply]
Using an external simple ping script seems reasonable, thanks for your input.
